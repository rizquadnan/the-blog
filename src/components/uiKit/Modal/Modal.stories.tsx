import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { Modal } from './Modal'
import { Button, useDisclosure } from '@chakra-ui/react'

type TModal = React.ComponentProps<typeof Modal>
export default {
  title: 'Overlay/Modal',
  component: Modal,
} as Meta<TModal>

const Template: Story<TModal> = (args) => {
  const { isOpen, onClose, onOpen } = useDisclosure()
  return (
    <>
      <Button onClick={onOpen}>Open Modal</Button>
      <Modal {...args} isOpen={isOpen} onClose={onClose} />
    </>
  )
}

export const Default = Template.bind({})
Default.args = {
  body: (
    <div>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem, cumque.
      Eaque commodi fuga eligendi velit molestiae beatae quibusdam magni, et
      suscipit? Porro facere enim vel ad pariatur. Sed, temporibus praesentium.
    </div>
  ),
  title: 'Modal Title',
}
