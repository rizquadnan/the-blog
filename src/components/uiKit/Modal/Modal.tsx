import {
  Modal as _Modal,
  ModalProps as _ModalProps,
  ModalBody,
  ModalBodyProps,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from '@chakra-ui/react'
import React, { ReactNode } from 'react'

type ModalProps = Omit<_ModalProps, 'children'> & {
  body: ReactNode
  title: string
  bodyContainerProps?: ModalBodyProps
  footer?: ReactNode
}

export function Modal({ bodyContainerProps = {}, ...props }: ModalProps) {
  return (
    <_Modal size={{ base: 'full', xl: '3xl' }} {...props}>
      <ModalOverlay />
      <ModalContent>
        <ModalHeader fontSize="32px" fontWeight={800} color="orange.600">
          {props.title}
        </ModalHeader>
        <ModalCloseButton size="lg" />
        <ModalBody {...bodyContainerProps}>{props.body}</ModalBody>
        {props.footer !== undefined && (
          <ModalFooter display="block">{props.footer}</ModalFooter>
        )}
      </ModalContent>
    </_Modal>
  )
}

export default Modal
