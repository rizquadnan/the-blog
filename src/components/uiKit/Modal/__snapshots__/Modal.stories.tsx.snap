// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay/Modal Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Open Modal",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "overlay-modal--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
